diff --git a/cyber/data/channel_buffer_test.cc b/cyber/data/channel_buffer_test.cc
index 98ea296939..8bff226bb1 100644
--- a/cyber/data/channel_buffer_test.cc
+++ b/cyber/data/channel_buffer_test.cc
@@ -27,7 +27,7 @@ namespace apollo {
 namespace cyber {
 namespace data {
 
-auto channel0 = common::Hash("/channel0");
+static auto channel0 = common::Hash("/channel0");
 
 TEST(ChannelBufferTest, Fetch) {
   auto cache_buffer = new CacheBuffer<std::shared_ptr<int>>(2);
diff --git a/cyber/data/data_dispatcher_test.cc b/cyber/data/data_dispatcher_test.cc
index 16cf797a77..f159259ecd 100644
--- a/cyber/data/data_dispatcher_test.cc
+++ b/cyber/data/data_dispatcher_test.cc
@@ -31,8 +31,8 @@ template <typename T>
 using BufferVector =
     std::vector<std::weak_ptr<CacheBuffer<std::shared_ptr<T>>>>;
 
-auto channel0 = common::Hash("/channel0");
-auto channel1 = common::Hash("/channel1");
+static auto channel0 = common::Hash("/channel0");
+static auto channel1 = common::Hash("/channel1");
 
 TEST(DataDispatcher, AddBuffer) {
   auto cache_buffer1 = new CacheBuffer<std::shared_ptr<int>>(2);
diff --git a/cyber/data/data_visitor_test.cc b/cyber/data/data_visitor_test.cc
index 2353bb2054..24213315a7 100644
--- a/cyber/data/data_visitor_test.cc
+++ b/cyber/data/data_visitor_test.cc
@@ -32,12 +32,12 @@ namespace data {
 
 using apollo::cyber::message::RawMessage;
 using apollo::cyber::proto::RoleAttributes;
-std::hash<std::string> str_hash;
+static std::hash<std::string> str_hash;
 
-auto channel0 = str_hash("/channel0");
-auto channel1 = str_hash("/channel1");
-auto channel2 = str_hash("/channel2");
-auto channel3 = str_hash("/channel3");
+static auto channel0 = str_hash("/channel0");
+static auto channel1 = str_hash("/channel1");
+static auto channel2 = str_hash("/channel2");
+static auto channel3 = str_hash("/channel3");
 
 void DispatchMessage(uint64_t channel_id, int num) {
   for (int i = 0; i < num; ++i) {
diff --git a/cyber/data/fusion/all_latest_test.cc b/cyber/data/fusion/all_latest_test.cc
index ce8fb69fd3..435041f150 100644
--- a/cyber/data/fusion/all_latest_test.cc
+++ b/cyber/data/fusion/all_latest_test.cc
@@ -30,7 +30,7 @@ namespace data {
 
 using apollo::cyber::message::RawMessage;
 using apollo::cyber::proto::RoleAttributes;
-std::hash<std::string> str_hash;
+static std::hash<std::string> str_hash;
 
 TEST(AllLatestTest, two_channels) {
   auto cache0 = new CacheBuffer<std::shared_ptr<RawMessage>>(10);
